### PR TITLE
Security fix of OTR, PGP and other

### PR DESCRIPTION
**Pull request by H3rnand3zzz**
Fix of OTR in message.c: carbon in stanza breaks OTR and prevents it from starting and working properly

Fix of other encryption methods by calling correct function that would initiate them in case if someone writes.

PGP Bug Description
If someone writes using PGP and chat window was closed even if PGP was enabled before, PGP won't initiate, even though it should. The problem is in calling the wrong function to initiate window. It can make a user send a message which he expect to be encrypted in plain text, thus making a big security problem.

OTR Bug Description
If user tries to initiate OTR, it does not work. E.g. PSI+ will just print out the messages and other profanity client won't start the OTR session either. The problem is in the carbons inside of the message.

Minor formatting.

### CODE REVIEW

File: src/config/color.c [hunk 1]
```diff
@@ -59,7 +59,7 @@ static struct color_pair_cache
     struct
     {
         int16_t fg, bg;
-    } * pairs;
```
Comment (jubalh) [inline src/config/color.c:62]: For this please see https://github.com/profanity-im/profanity/pull/1774

File: src/xmpp/message.c [hunk 2]
```diff
@@ -598,10 +598,6 @@ message_send_chat_otr(const char* const barejid, const char* const msg, gboolean
         stanza_attach_state(ctx, message, state);
     }
 
-    stanza_attach_carbons_private(ctx, message);
```
Comment (jubalh) [inline src/xmpp/message.c:601]: Please move the OTR change in a separate commit. So that you have one commit per change that belongs together.

**Review by jubalh — CHANGES_REQUESTED** [review]
Thanks for you patch! I didn't review it in detail yet. Only superficial. Please see the comments.

File: src/xmpp/message.c (hunk 2 above)
Comment (H3rnand3zzz) [reply src/xmpp/message.c:601]: Done.

File: src/config/color.c (hunk 1 above)
Comment (H3rnand3zzz) [reply src/config/color.c]: Done.

File: src/xmpp/message.c (hunk 2 above)
Comment (jubalh) [reply src/xmpp/message.c:601]: Ah yes :) But now we have 1) the not optimal commit 2) the revert commit 3) one good commit 4) another good commit. And the last 2 we actually don't need in the history. You can drop the first two commits or ammend them. Using something like `git rease -i HEAD~5`. Maybe google the details. It's quite handy to know about it. Afterwards you will have to do a `git push -f` to force push it on this branch.

**Review by jubalh — CHANGES_REQUESTED** [review]
Ah yes :) But now we have 1) the not optimal commit 2) the revert commit 3) one good commit 4) another good commit. And the last 2 we actually don't need in the history. You can drop the first two commits or ammend them. Using something like `git rease -i HEAD~5`. Maybe google the details. It's quite handy to know about it. Afterwards you will have to do a `git push -f` to force push it on this branch.

**Review by sjaeckel — CHANGES_REQUESTED** [review]
Thanks for those patches!

Could you please rebase your changes and remove the first two commits (initial commit and its revert)?
How did you find those problems (if I may ask)?

After checking for other references of `wins_new_chat()` I've found that it's still used from `src/otr/otr.c:150 cb_gone_secure()` ... should this maybe also be changed to calling `chatwin_new()` instead?

**Comment by H3rnand3zzz** [comment]
> Thanks for those patches!
> 
> Could you please rebase your changes and remove the first two commits (initial commit and its revert)? How did you find those problems (if I may ask)?
> 
> After checking for other references of `wins_new_chat()` I've found that it's still used from `src/otr/otr.c:150 cb_gone_secure()` ... should this maybe also be changed to calling `chatwin_new()` instead?

Thanks for the feedback. I found these problems when I was trying to communicate with other people using profanity. Fixed it.

**Comment by H3rnand3zzz** [comment]
> Ah yes :) But now we have 1) the not optimal commit 2) the revert commit 3) one good commit 4) another good commit. And the last 2 we actually don't need in the history. You can drop the first two commits or ammend them. Using something like `git rebase -i HEAD~5`. Maybe google the details. It's quite handy to know about it. Afterwards you will have to do a `git push -f` to force push it on this branch.

Thank you for the feedback. I knew about this feature, but wasn't sure that "push -f" is a good practice. Now I learned that under certain circumstances, it can be used for the greater good.

**Review by sjaeckel — APPROVED** [review]

**Comment by jubalh** [comment]
> but wasn't sure that "push -f" is a good practice

That's true if several people work on the same branch :)

**Review by jubalh — APPROVED** [review]

**Comment by jubalh** [comment]
Thanks for your contribution @H3rnand3zzz !

**Comment by sjaeckel** [comment]
> > but wasn't sure that "push -f" is a good practice
> 
> That's true if several people work on the same branch :)

FYI: Since some time there are the [--force-with-lease](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-force-with-lease) and [--force-if-includes](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-force-if-includes) options for `git push` which are supposed to prevent the most common mishaps when working together on the same branch.
